CommonMark/Formatters/HtmlFormatter.cs

```diff
@@ -15,6 +15,7 @@ public class HtmlFormatter
         private readonly CommonMarkSettings _settings;
         private readonly Stack<bool> _renderTightParagraphs = new Stack<bool>(new[] { false });
         private readonly Stack<bool> _renderPlainTextInlines = new Stack<bool>(new[] { false });
+        private readonly Stack<char?> _endPlaceholders = new Stack<char?>();
```

Comment:
Do not use `char?` - instead use a special value for the `null` case such as `\0` - this will save a little memory and be very slightly faster